### ISSUE TITLE
Identation rules are broken

### ISSUE THREAD

**Issue by fabiospampinato** [microsoft/vscode#57197]
- VSCode Version: v1.26.1
- OS Version: macOS 10.13.5

I think I've had the following problems since the very first version of VSC I tried.

### Problem 1

![up](https://user-images.githubusercontent.com/1812093/44618558-91c5c780-a878-11e8-9721-022a87a45b59.gif)

1. Have [Todo+](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-todo-plus) installed and create a new "todo" document with the same content as mine.
2. Move the line up.
3. Undo.
4. Move the line up.
5. We get 2 different results for doing the same thing.

### Problem 2 

![down](https://user-images.githubusercontent.com/1812093/44618582-1b759500-a879-11e8-8b48-28882a0bc20d.gif)

1. Have [Todo+](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-todo-plus) installed and create a new "todo" document with the same content as mine.
2. Move the line down.
3. Undo.
4. Move the line down.
5. We get 2 different results for doing the same thing.

### Problem 3

In both the gifs above the indentation for a line gets decreased, my understanding is that this is governed by the [`indentationRules.decreaseIndentPattern`](https://github.com/fabiospampinato/vscode-todo-plus/blob/master/src/utils/init.ts/#L38-L38) setting, which in my extension is set to `/(?=a)b/gm`, a regex that can never match any string, why are those lines' indentations decreased then?

**Comment by marco-m** [comment]
Hello @rebornix, any chance to fix this bug ? Thanks!

**Comment by roblourens** [comment]
If I start with this

```
Project: (2)
	☐ hi
	☐ hi2
```

Then "move line up" on the second line, I get this

```
Project: (2)
☐ hi2
	☐ hi
```

Is that what's expected?

**Comment by marco-m** [comment]
Thanks @rebornix :-)

**Comment by fabiospampinato** [comment]
> Is that what's expected?

@roblourens No, in that scenario the task being moved up remains on the same level so its indentation shouldn't change.
